src/serde_util.rs

```diff
@@ -67,3 +67,44 @@ where
         (self.parse_fn)(v).map_err(E::custom)
     }
 }
+
+pub struct SeqVisitor<F> {
+    expectation: &'static str,
+    parse_fn: F,
+}
+
+impl<F, T, Err> SeqVisitor<F>
+where
+    F: FnOnce(&[u8]) -> Result<T, Err>,
+    Err: fmt::Display,
+{
+    pub fn new(expectation: &'static str, parse_fn: F) -> Self {
+        SeqVisitor {
+            expectation,
+            parse_fn,
+        }
+    }
+}
+
+impl<'de, F, T, Err> de::Visitor<'de> for SeqVisitor<F>
+where
+    F: FnOnce(&[u8]) -> Result<T, Err>,
+    Err: fmt::Display,
+{
+    type Value = T;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str(self.expectation)
+    }
+
+    fn visit_seq<V>(self, mut seq: V) -> Result<Self::Value, V::Error>
+    where
+        V: de::SeqAccess<'de>,
+    {
+        let mut v = Vec::new();
```

Comment:
We should probably tell the `Vec` using `with_capacity`, how many elements are coming: https://docs.rs/serde/1.0.136/serde/de/trait.SeqAccess.html#method.size_hint

Perhaps doing some `criterion` benchmarks could be useful to measure the performance impact of this change?

Reply:
Good idea.

Reply:
We do not know the size at this stage since public key and secret key are different length, if we choose to do fixed width for other types there will be even more uncertainty.

Reply:
You can use the API I linked to get the size :)

Reply:
duh, sometimes my own stupidity amazes me, can do nothing but chuckle at myself.

Reply:
```
        #[cfg(all(not(feature = "std"), feature = "alloc"))]
        type Vec = alloc::vec::Vec;

        let mut v = match seq.size_hint() {
            Some(size) => Vec::with_capacity(size),
            None => Vec::new(),
        };
```